edb/server/pgcon/pgcon.pyx

```diff
@@ -570,19 +570,26 @@ cdef class PGConnection:
             await self.after_command()
 
     async def wait_for_sync(self):
+        error = None
         while True:
             if not self.buffer.take_message():
                 await self.wait_for_message()
             mtype = self.buffer.get_message_type()
             if mtype == b'Z':
                 self.parse_sync_message()
-                return
+                break
```

Comment:
OK, this is subtle. Please put the entire `while True` in a `try..finally`, with `finally` checking on `error is not None`. 

This will prevent us from adding another return (and not break) somewhere inside this block in the future. Also, please add a comment to the finally block explaining the current conundrum.

Reply:
`try..finally` would also save us from an unrelated Python-level exception masking the more important and potentially trickier Postgres error.